src/librustc_ast_lowering/lib.rs

```diff
@@ -596,11 +596,11 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
             hir_id
         };
 
-        if !span.is_dummy() {
-            match self.spans.insert(hir_id, span) {
-                Some(sp) if !sp.is_dummy() => assert_eq!(sp, span),
-                _ => {}
-            };
+        let stored_span = self.spans.entry(hir_id).or_insert(span);
+        if *stored_span == DUMMY_SP {
```

Comment:
Why not keep using `is_dummy()` here? (And `!is_dummy()` below?)

I don't have a preference personally, but I also wonder when I see a change from an existing code style like this.

Reply:
(this particular instance of comparison with `DUMMY_SP` was removed in a follow-up commit. But the question still stands over all, since the PR in general does inject more comparisons with `DUMMY_SP`, for better or for worse.)

Reply:
In some cases, the HIR owners are assigned a HirId before the AST traversal. When this HirId is created, we do not have access to the proper Span. I use DUMMY_SP as a placeholder for delayed initialisation. I will add a comment about that.